### PR TITLE
Global træfik proxy

### PR DESCRIPTION
**Pull request by cableman**
Ticket: https://jira.itkdev.dk/browse/TEAMMANAGE-1536

This will enable us to have a docker-compose setup with one træfik project as ingress and don't have to use port numbers and domain names will be free.

Screenshot 2019-09-26 at 22 22 49

@TODO: make the traefik containers start automatically

### CODE REVIEW

File: templates/ddbcms/docker-compose.yml [hunk 1]
```diff
@@ -30,32 +41,43 @@ services:
...
+    ports:
+      - '80'
     volumes:
       - ${PWD}/.docker/vhost.conf:/etc/nginx/conf.d/default.conf:ro
       - ./:/app:delegated
     labels:
-      - "traefik.http.routers.frontend.rule=Host(`${COMPOSE_PROJECT_NAME}.docker.localhost`)"
+      - "traefik.enable=true"
+      - "traefik.docker.network=frontend"
+      - "traefik.http.routers.frontend.rule=Host(`${${COMPOSE_DOMAIN}}`)"
```
[the remark this answers is not in the document]
Comment (cableman) [reply templates/ddbcms/docker-compose.yml]: Copy & replace error.  Fixed

File: templates/ddbcms/docker-compose.yml [hunk 2]
```diff
@@ -21,7 +32,7 @@ services:
       - PHP_MAX_EXECUTION_TIME=300
       - PHP_MEMORY_LIMIT=512M
       # - PHP_MAIL=1 # Uncomment to enable mailhog.
-      - DOCKER_HOST_DOMAIN=${COMPOSE_PROJECT_NAME}.docker.localhost
+      - DOCKER_HOST_DOMAIN=${${COMPOSE_DOMAIN}}
```
[the remark this answers is not in the document]
Comment (cableman) [reply templates/ddbcms/docker-compose.yml]: Copy & replace error.  Fixed

File: scripts/itkdev-docker-compose [hunk 3]
```diff
@@ -145,7 +142,9 @@ if [ -z "${COMPOSE_PROJECT_NAME:-}" ]; then
   exit 1
 fi
 
-COMPOSE_DOMAIN=${COMPOSE_DOMAIN:-${COMPOSE_PROJECT_NAME}.docker.localhost}
+# Needs to be exported as it's used in docker-compose.yml and if not set in .env it will be the 
+# empty string, hence default to *.docker.localhost will not work.
+export COMPOSE_DOMAIN=${COMPOSE_DOMAIN:-${COMPOSE_PROJECT_NAME}.docker.localhost}
```
[the remark this answers is not in the document]
Comment (cableman) [reply scripts/itkdev-docker-compose]: agree, just need to figure out how

[the remark this answers is not in the document]
Comment (cableman) [reply scripts/itkdev-docker-compose]: Fixed

File: scripts/itkdev-docker-compose [hunk 4]
```diff
@@ -47,6 +49,15 @@ Commands:
   sql:port
                 Display the exposed MariaDB SQL server port.
 
+  traefik:start
+                Start træfik reserver proxy.
+
+  traefik:stop
+                Stop træfik reverser proxy
+
+  traefik:open
```
Comment (rimi-itk) [inline scripts/itkdev-docker-compose]: For consistency with other similar commands, we should add `traefik:url` to the mix.

File: completion/zsh/_itkdev-docker-compose [hunk 5]
```diff
@@ -35,6 +35,9 @@ _itkdev-docker-compose() {
               'sync\:files[Sync files base.]' \
               'sql\:connect[Print mysql command for connecting to database.]' \
               'sql\:port[Display the exposed MariaDB SQL server port.]' \
+              'traefik\:start[Start reverser proxy]' \
```
[the remark this answers is not in the document]
Comment (cableman) [reply completion/zsh/_itkdev-docker-compose]: yes, fixed

File: scripts/itkdev-docker-compose [hunk 6]
```diff
@@ -145,6 +153,8 @@ if [ -z "${COMPOSE_PROJECT_NAME:-}" ]; then
   exit 1
 fi
 
+# Needs to be exported as it's used in docker-compose.yml and if not set in .env it will be the
```
[the remark this answers is not in the document]
Comment (cableman) [reply scripts/itkdev-docker-compose]: Removed

File: scripts/itkdev-docker-compose [hunk 7]
```diff
@@ -4,6 +4,8 @@ IFS=$'\n\t'
 
 script_dir=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
 script_path=$script_dir/$(basename "${BASH_SOURCE[0]}")
+script_real_path=$(readlink /usr/local/bin/itkdev-docker-compose)
```
[the remark this answers is not in the document]
Comment (cableman) [reply scripts/itkdev-docker-compose]: Fixed with helper function to handle both symlinked and in path

File: scripts/itkdev-docker-compose (hunk 4 above)
Comment (cableman) [reply scripts/itkdev-docker-compose]: Added

**Pull request merged**